### PR TITLE
ci: add build and test workflows

### PR DESCRIPTION
**Pull request by Xunzhuo**
**What type of PR is this?**

**What this PR does / why we need it**:

- [ ] Build and Test Workflow: 
  - [ ] runs test: unit test, e2e test(TODO) of i2gw.
  - [ ] runs linters: golint, yamllint, codespells.
  - [ ] runs multi-arch build of i2gw.

Sub of #9

**Which issue(s) this PR fixes**:

xref: #21

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by Xunzhuo** [comment]
xref: https://github.com/kubernetes-sigs/ingress2gateway/issues/21

File: .github/workflows/build_and_test.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    paths-ignore:
+    - "**/*.png"
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: ./tools/github-actions/setup-deps
+    - run: make lint-deps
+    - run: make -k lint
```
Comment (robscott) [inline .github/workflows/build_and_test.yaml:36]: Which PR adds these?

**Review by robscott — COMMENTED** [review]
Thanks for working on this @Xunzhuo! Do you mind linking which PR(s) need to merge before this?

/hold

**Comment by robscott** [comment]
@Xunzhuo let us know if you want to follow up on these, for now I'm going to close out in favor of more recent PRs.

**Pull request closed without merging**